src/server/schema.js

```diff
@@ -42,60 +47,154 @@ const gqlTypeToAggsHistogramName = {
 
 const getAggsHistogramName = (gqlType) => {
   if (!gqlTypeToAggsHistogramName[gqlType]) {
-    throw new Error(`Invalid elasticsearch type ${gqlType}`);
+    // throw new Error(`Invalid elasticsearch type ${gqlType}`);
+    return '';
   }
   return gqlTypeToAggsHistogramName[gqlType];
 };
 
-const getQuerySchemaForType = (esType) => {
+const getQuerySchemaForType = (esInstance, esIndex, esType) => {
   const esTypeObjName = firstLetterUpperCase(esType);
-  return `${esType} (
+  const fieldESTypeMap = esInstance.getESFieldTypeMappingByIndex(esIndex);
+  const existingFields = new Set([]);
+
+  const queueFields = [];
+  Object.keys(fieldESTypeMap).forEach((fieldKey) => {
+    const esFieldType = fieldESTypeMap[fieldKey].type;
+    if (esFieldType === 'nested' && !existingFields.has(fieldKey)) {
+      queueFields.push({ key: fieldKey, value: fieldESTypeMap[fieldKey] });
+      existingFields.add(fieldKey);
+    }
+  });
+  let sType = `${esType} (
     offset: Int, 
     first: Int,
     filter: JSON,
     sort: JSON,
     accessibility: Accessibility=all,
     ): [${esTypeObjName}]`;
+  while (queueFields.length > 0) {
+    const f = queueFields.shift();
+    sType += `
+
+    ${f.key} (
+      offset: Int,
+      first: Int,
+      filter: JSON,
+      sort: JSON,
+      accessibility: Accessibility=all,
+    ):  [${firstLetterUpperCase(f.key)}]`;
+    Object.keys(f.value.properties).forEach((p) => {
+      if (!existingFields.has(p)) {
+        const field = f.value.properties[p];
+        if (field.type === 'nested') {
+          queueFields.push({ key: p, value: field });
+          existingFields.add(p);
+        }
+      }
+    });
+  }
+  return `${sType}
+  `;
 };
 
+// eslint-disable-next-line max-len
```

Comment:
let's not ignore `max-len` warning, could we do wrapping? like how everywhere else in this repo dealing with eslint